### PR TITLE
Fix the pre aggregation is difference between old and new selector

### PR DESCRIPTION
**Pull request by EmmyMiao87**
The issue is #3016.
If there is no aggregated column in aggregate index, the index will be deduplicate table.
For example:
    aggregate table (k1, k2, v1 sum)
    mv index (k1, k2)
This kind of index is SPJG which same as select k1, k2 from aggregate_table group by k1, k2.
It also need to check the grouping column using following steps.

If there is no aggregated column in duplicate index, the index will be SPJ which passes the grouping verification directly.

Also after the supplement of index, the new candidate index should be checked the output columns also.

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/planner/MaterializedViewSelector.java [hunk 1]
```diff
@@ -279,16 +280,30 @@ private void checkCompensatingPredicates(Set<String> columnsInPredicates,
...
+                mv index (k1, v1)
+            When the candidate index is SPJ type, it passes the verification directly
+
+            If there is no aggregated column in aggregate index, the index will be deduplicate table.
+            For example:
+                aggregate table (k1, k2, v1 sum)
+                mv index (k1, k2)
+            This kind of index is SPJG which same as select k1, k2 from aggregate_table group by k1, k2.
+            It also need to check the grouping column using following steps.
+             */
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/planner/MaterializedViewSelector.java]: Please add a UT to cover this case.

Comment (EmmyMiao87) [reply fe/src/main/java/org/apache/doris/planner/MaterializedViewSelector.java]: OK

**Review by morningman — APPROVED** [review]
LGTM

**Comment by EmmyMiao87** [comment]
#2101

**Pull request merged**